example/src/Booking/Api.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Booking.Api where
+
+import           Booking.Types
+import           Data.Set      (Set)
+import           Data.Text     (Text)
+import           Servant
+
+
+type BookingAPI = "hotelbooking" :>
```

Comment:
This API is not well defined according to the rules of nesting: we want that "/hotelBooking/" is present in all URL's so we you nest all endpoints inside it, that is:
```
type BookingAPI = "hotelbooking" :>
       -- ^ 1) Return a set of availables rooms based on the CheckInOut dates                                                                                                                                                                  
  (    "getAvailableRooms" :> ReqBody '[JSON] CheckInOut :> Get '[JSON] (Set Room)

       -- ^ 2) Return a bolean if the given RoomId is available                                                                                                                                                                                
  :<|> "isRoomAvailable":> ReqBody '[JSON] Text :> Get '[JSON] Bool

       -- ^ 3) Return a report set based on a set of CheckInOut dates                                                                                                                                                                          
  :<|> "getPeriodicReport" :> ReqBody '[JSON] (Set CheckInOut) :> Get '[JSON] (Set Report)

       -- ^ 4) Creates a reservation based on a Reservation data                                                                                                                                                                               
  :<|> "booking" :> ReqBody '[JSON] Reservation :> Post '[JSON] Reservation

       -- ^ 5) Removes a reservation based on the ReservationId                                                                                                                                                                                
  :<|> "cancel" :> ReqBody '[JSON] Text :> Post '[JSON] Reservation
  )
```
Remember that the  `:<|>`  type-operator is for specifying a new endpoint and the `:>` type-operator is for specifying a part of an end-point.
So when we write  `"hotelbooking" :> ( ... )` we want that all endpoints inside the parenthesis are nested inside the "hotelBooking" segment...

:)